docs/design/features/lsra-detail.md

```diff
@@ -928,6 +1088,40 @@ and we may want to experiment with changing the allocation order as
 well. It would not be necessary to significantly modify the process of
 creating `Interval`s, nor the resolution phase.
 
+## Throughput Enhancements
+
+### Allocation Window for Min-Opts and Tier 0
+
+Currently, the register allocator builds the RefPositions for the entire method at once.
+In Min-Opts and Tier 0, or any time there are no local variables available for enregistration,
+there is no value in doing so, as no values remain live beyond the top-level node of an
+expression. Although it is possible for the nodes of multiple top-level trees to overlap each
```

Comment:
> Although it is possible for the nodes of multiple top-level trees to overlap each other in execution order, that is not commonly the case.

Could you please clarify that statement for me? Does it mean that live intervals for the root trees before Lowering usually don't overlap ?

Reply:
That's right. Since we don't do instruction scheduling or other reordering, it is generally the case that the nodes for disjoint top-level trees do not overlap.